modules/generator/instance.go

```diff
@@ -246,6 +258,14 @@ func (i *instance) updateProcessors() error {
 // diffProcessors compares the existing processors with the desired processors and config.
 // Must be called under a read lock.
 func (i *instance) diffProcessors(desiredProcessors map[string]struct{}, desiredCfg ProcessorConfig) (toAdd, toRemove, toReplace []string, err error) {
+	var reloadForHistograms bool
+
+	currentHistogramsOverrides := i.overrides.MetricsGeneratorGenerateNativeHistograms(i.instanceID)
+	previousHisograms := i.histogramsOverride.Load()
+	if currentHistogramsOverrides == "classic" && currentHistogramsOverrides != previousHisograms {
```

Comment:
Maybe add a small comment why we only reload if `currentHistogramsOverrides == "classic"`. Might not be obvious for everyone.